src/routes/panel/routes.php

```diff
@@ -231,6 +231,128 @@
 
 });
 
+$klein->respond('GET', '/bulkcmd', function($request, $response, $service) use($core) {
+
+	$servers = ORM::forTable('servers')
+                        ->select('servers.*')->select('nodes.name', 'node_name')->select('locations.long', 'location')
+                        ->join('nodes', array('servers.node', '=', 'nodes.id'))
+                        ->join('locations', array('nodes.location', '=', 'locations.id'))
+                        ->where('servers.active', 1)
+                        ->where_in('servers.id', $core->permissions->listServers())
+                        ->findArray();
+
+	$actualServers = array();
+
+	// Sorry, I'm not sure how to get servers where they are the owner only!
+	foreach ($servers as $server) {
+		if($server['owner_id'] === $core->user->getData('id')) {
+			$actualServers[] = $server;
+		}
+	}
+
+	$response->body($core->twig->render('panel/bulkcmd.html', array(
+		'servers' => $actualServers,
+		'xsrf' => $core->auth->XSRF(),
+		'flash' => $service->flashes()
+	)))->send();
+
+});
+
+$klein->respond('POST', '/bulkcmd', function($request, $response, $service) use ($core) {
+
+	if(!$core->auth->XSRF($request->param('xsrf'))) {
+
+                $service->flash('<div class="alert alert-warning">'.$core->language->render('error.xsrf').'</div>');
+                $response->redirect('/bulkcmd')->send();
+                return;
+
+        }
+
+	$servers = ORM::forTable('servers')
+                        ->select('servers.*')->select('nodes.name', 'node_name')->select('locations.long', 'location')
+                        ->join('nodes', array('servers.node', '=', 'nodes.id'))
+                        ->join('locations', array('nodes.location', '=', 'locations.id'))
+                        ->where('servers.active', 1)
+                        ->where_in('servers.id', $core->permissions->listServers())
+                        ->findArray();
+
+        $actualServers = array();
+
+	// Sorry, I'm not sure how to get servers where they are the owner only!
```

Comment:
Ah, okay. Your message there really did not read like that, and that kind of limitation is good with me.